shell/platform/android/io/flutter/plugin/editing/SpellCheckPlugin.java

```diff
@@ -0,0 +1,134 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugin.editing;
+
+import android.view.textservice.SentenceSuggestionsInfo;
+import android.view.textservice.SpellCheckerSession;
+import android.view.textservice.SuggestionsInfo;
+import android.view.textservice.TextInfo;
+import android.view.textservice.TextServicesManager;
+import androidx.annotation.NonNull;
+import io.flutter.embedding.engine.systemchannels.SpellCheckChannel;
+import io.flutter.plugin.localization.LocalizationPlugin;
+import java.util.ArrayList;
+import java.util.Locale;
+
+/**
+ * {@link SpellCheckPlugin} is the implementation of all functionality needed for spell check for
+ * text input.
+ *
+ * <p>The plugin handles requests for spell check sent by the {@link
+ * io.flutter.embedding.engine.systemchannels.SpellCheckChannel} via sending requests to the Android
+ * spell checker. It also receive the spell check results from the service and send them back to the
+ * framework through the {@link io.flutter.embedding.engine.systemchannels.SpellCheckChannel}.
+ */
+public class SpellCheckPlugin
+    implements SpellCheckChannel.SpellCheckMethodHandler,
+        SpellCheckerSession.SpellCheckerSessionListener {
+
+  private final SpellCheckChannel mSpellCheckChannel;
+  private final TextServicesManager mTextServicesManager;
+  private SpellCheckerSession mSpellCheckerSession;
+
+  // The maximum number of suggestions that the Android spell check service is allowed to provide
+  // per word.
```

Comment:
Just double checking, but are you sure that this is per word and not total?

Reply:
Yes!

Reply:
Ok cool, I thought the same thing when reading the documentation.